x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -47,7 +47,10 @@ func (k Keeper) SettlePendingClaims(ctx cosmostypes.Context) (
 	logger.Debug("settling expiring claims")
 	numExpiringClaims := 0
 	for ; expiringClaimsIterator.Valid(); expiringClaimsIterator.Next() {
-		claim := expiringClaimsIterator.Value()
+		claim, iterErr := expiringClaimsIterator.Value()
+		if iterErr != nil {
+			return settledResults, expiredResults, iterErr
+		}
 		numExpiringClaims++
```

Comment:
Do we not need to check if `claim` is nil, just in case?

Reply:
As mentioned in other comments, I will revert to return values instead of pointers.

A nil positive check will trigger an error as we should not expect a nil claim stored onchain.